File: Php/poo/salary/library/roster.php
<?php

    class Roster {
        
    const SALARY_MINIMUM = 1300000;
    const SUBSIDY_TRANSPORTATION = 140650;

    private Person $person;
    private Discount $discount;
    private float $subsidy = 0.0;
    private float $totalPay = 0.0;

    public function __construct(Person $person, Discount $discount) {
        $this->person = $person;
        $this->discount = $discount;
    }

    public function calculatePay(): float {
        $salary = $this->person->calculateSalary();
        $totalDiscount = $this->discount->calculateTotal();

        $this->subsidy = ($salary < 2 * self::SALARY_MINIMUM) ? self::SUBSIDY_TRANSPORTATION : 0;
        $this->totalPay = ($salary + $this->subsidy) - $totalDiscount;

        return $this->totalPay;
    }

    public function getSubsidy(): float {
        return $this->subsidy;
    }

    public function getTotalPay(): float {
        return $this->totalPay;
    }
}

File: Java/poo/salary/library/roster.java
package poo.salary.library;

public class roster {

    public static final double SALARY_MINIMUM = 1300000;
    public static final double SUBSIDY_TRANSPORTATION = 140650;

    private double salary;
    private double totalDiscount;
    private double subsidy = 0.0;
    private double totalPay = 0.0;

    public roster(double salary, double totalDiscount) {
        this.salary = salary;
        this.totalDiscount = totalDiscount;
    }

    public double calculatePay() {
        subsidy = (salary < 2 * SALARY_MINIMUM) ? SUBSIDY_TRANSPORTATION : 0;
        totalPay = (salary + subsidy) - totalDiscount;
        return this.totalPay;
    }

    public double getSubsidy() {
        return this.subsidy;
    }

    public double getTotalPay() {
        return (this.getTotalPay());
    }
}

